extensions/modulation.sigmf-ext.md

```diff
@@ -0,0 +1,196 @@
+5234](https://tools.ietf.org/html/rfc5234) and updated by [RFC
+7405](https://tools.ietf.org/html/rfc7405).
+
+Fields defined as "human-readable", a "string", or simply as "text" shall be
+treated as plaintext where whitespace is significant, unless otherwise
+specified.
+
+## Specification
+
+This specification defines an extension `modulation` namespace
+that can be used in SigMF `annotations` to describe wireless communications
+systems.
+
+### Annotations
+
+This extension adds the following field to the `annotations` global SigMF object:
+
+|name|required|type|description|
+|----|--------------|-------|-----------|
+|modulation|false|object|Describes a communications system.|
+
+The field of communications is vast, and there may be communications systems that cannot be described using the names and fields described in this extension. If you need additional or different fields to describe a system, create a new extension that adds the necessary fields to the `modulation` namespace and/or submit the new fields to be upstreamed into this specification.
+
+#### The `modulation` Object
+
+`modulation` objects contain name/value pairs that describe the modulations of communications systems. These objects MAY contain the following pairs:
+
+|name|required|type|
+|----|--------------|-------|
+|`type`|false|string|
+|`class`|false|string|
+|`carrier_variant`|false|string|
+|`symbol_variant`|false|string|
+|`order`|false|uint|
+|`duplexing`|false|string|
+|`multiplexing`|false|string|
+|`multiple_access`|false|string|
+|`spreading`|false|string|
+|`bandwidth`|false|float|
+|`system`|false|string|
+
+###### The `type` Pair
+
+The `type` name can have the following values:
+
+|value|description|
+|----|-------|
+|`analog`|analog modulation scheme|
+|`digital`|digital modulation scheme|
+
+###### The `class` Pair
+
+The `class` name can have the following values:
+
+|value|description|
+|----|-------|
+|`am`|(analog) amplitude modulation|
+|`fm`|(analog) frequency modulation|
+|`pm`|(analog) phase modulation|
+|`ssb`|(analog) single side-band|
+|`dsb`|(analog) dual side-band|
+|`vsb`|(analog) vestigal side-band|
+|`ask`|(analog) amplitude-shift keying|
+|`fsk`|(digital) frequency-shift keying|
+|`psk`|(digital) phase-shift keying|
+|`qam`|(digital) quadrature-amplitude modulation|
+|`ook`|(digital) on-off keying|
+|`cpm`|(digital) continuous phase modulation|
+|`msk`|(digital) minimum-shift keying|
+
+###### The `carrier_variant` pair
+
+The `carrier_variant` name can have the following values:
+
+|value|type|description|
+|----|----|-------|
+|`with_carrier`|(analog) with-carrier modulation|
+|`suppressed_carrier`|(analog) suppressed-carrier modulation|
+|`reduced_carrier`|(analog) reduced-carrier modulation|
+|`single_carrier`|(digital) single-carrier modulation|
+|`multi_carrier`|(digital) multi-carrier modulation|
+
+###### The `carrier_variant` pair
+
+The `symbol_variant` name can have the following values:
```

Comment:
The section heading should probably be `symbol_variant`, but this is another key that I'm not sure makes 100% sense. For the offset case, this is just another modulation, (it's OQPSK or it's QPSK). For the differential case, to be fair you can't identify that at the PHY, so maybe it's just a true or false? It's hard to say since I don't have a practical need for declaring if a signal is differential or not, but I don't think a single value should declare differential OR offset (because they could be both)